src/jit/error.cpp

```diff
@@ -76,6 +84,12 @@ void DECLSPEC_NORETURN NOMEM()
     fatal(CORJIT_OUTOFMEM);
 }
 
+/*****************************************************************************/
+void DECLSPEC_NORETURN rejitTier0ToTier1()
```

Comment:
It is very unfortunate to use exception handling for this. Why can't it be done with regular control flow?

Reply:
These exceptions would be pain for us and confusing for our customers, e.g. they will show up in Visual Studio debugger.

Reply:
It looks like it would be ok to change the opt level and continue jitting. @AndyAyersMS had suggested this before, I had misread the order of some code before and thought the opt level was being used before the point at which the switch is made.